source/UF/Pushouts.lagda

```diff
@@ -0,0 +1,418 @@
+ pushout-fiber-is-singleton' {_} {P} t
+  = equiv-to-singleton'
+     (Σ-cong (λ - → dependent-cocone-identity-characterization f g pushout
+              pushout-cocone P (- ∘ inll , - ∘ inrr ,  λ c → apd - (glue c)) t))
+     (pushout-fiber-is-singleton t)
+
+ pushout-fiber-center
+  : {P : pushout →  𝓣'  ̇}
+  → (t : dependent-cocone f g pushout pushout-cocone P)
+  → Σ d ꞉ ((x : pushout) → P x) ,
+      dependent-cocone-family f g pushout pushout-cocone P
+       (d ∘ inll , d ∘ inrr ,  λ c → apd d (glue c)) t
+ pushout-fiber-center t = center (pushout-fiber-is-singleton' t)
+
+ pushout-fiber-centrality
+  : {P : pushout →  𝓣'  ̇}
+  → (t : dependent-cocone f g pushout pushout-cocone P)
+  → is-central (Σ d ꞉ ((x : pushout) → P x) ,
+                 dependent-cocone-family f g pushout pushout-cocone P
+                  (d ∘ inll , d ∘ inrr ,  λ c → apd d (glue c)) t)
+               (pushout-fiber-center t)
+ pushout-fiber-centrality t = centrality (pushout-fiber-is-singleton' t)
+
+ pushout-unique-map
+  : {P : pushout →  𝓣'  ̇}
+  → (t : dependent-cocone f g pushout pushout-cocone P)
+  → Σ d ꞉ ((x : pushout) → P x) ,
+     dependent-cocone-family f g pushout pushout-cocone P
+      (d ∘ inll , d ∘ inrr ,  λ c → apd d (glue c)) t
+  → (x : pushout) → P x
+ pushout-unique-map t (d , _) = d
+
+ pushout-inll-homotopy
+  : {P : pushout →  𝓣'  ̇}
+  → (t : dependent-cocone f g pushout pushout-cocone P)
+  → (z : Σ d ꞉ ((x : pushout) → P x) ,
+          dependent-cocone-family f g pushout pushout-cocone P
+           (d ∘ inll , d ∘ inrr ,  λ c → apd d (glue c)) t)
+  → (pushout-unique-map t z) ∘ inll
+  ∼ dependent-cocone-vertical-map f g pushout pushout-cocone P t
+ pushout-inll-homotopy s (u , K , L , M) = K
+
+ pushout-inrr-homotopy
+  : {P : pushout →  𝓣'  ̇}
+  → (t : dependent-cocone f g pushout pushout-cocone P)
+  → (z : Σ d ꞉ ((x : pushout) → P x) ,
+          dependent-cocone-family f g pushout pushout-cocone P
+           (d ∘ inll , d ∘ inrr ,  λ c → apd d (glue c)) t)
+  → (pushout-unique-map t z) ∘ inrr
+  ∼ dependent-cocone-horizontal-map f g pushout pushout-cocone P t
+ pushout-inrr-homotopy s (u , K , L , M) = L
+
+ pushout-glue-homotopy
+  : {P : pushout →  𝓣'  ̇}
+  → (t : dependent-cocone f g pushout pushout-cocone P)
+  → (z : Σ d ꞉ ((x : pushout) → P x) ,
+          dependent-cocone-family f g pushout pushout-cocone P
+           (d ∘ inll , d ∘ inrr ,  λ c → apd d (glue c)) t)
+  → ∼-trans (λ - → ap (transport P (glue -)) ((pushout-inll-homotopy t z ∘ f) -))
+            (dependent-cocone-commuting-square f g pushout pushout-cocone P t)
+  ∼ ∼-trans (λ - → apd (pushout-unique-map t z) (glue -))
+            ((pushout-inrr-homotopy t z) ∘ g)
+ pushout-glue-homotopy t (u , K , L , M) = M
+
+\end{code}
+
+Now we can derive the induction and recursion principles along with the corresponding
+computation rules, the uniqueness of maps out of the pushout and the (non-dependent)
+universal property.
+
+\begin{code}
```

Comment:
Again, in the code below, you should use anonymous modules for parameters that are used repeatedly by some functions.